src/main/scala/mesosphere/marathon/core/task/state/MarathonTaskStatus.scala

```diff
@@ -0,0 +1,100 @@
+package mesosphere.marathon.core.task.state
+
+import org.apache.mesos
+
+// TODO ju handle garbage flag
+sealed trait MarathonTaskStatus {
+  def toMesosStateName: String = "TASK_" + toString.toUpperCase
+}
+
+object MarathonTaskStatus {
+  import org.apache.mesos.Protos.TaskState._
+
+  def fromMesosStateName(mesosStateName: String): Option[MarathonTaskStatus] = all.collectFirst {
+    case status: MarathonTaskStatus if status.toMesosStateName == mesosStateName => status
+  }
+
+  // If we're disconnected at the time of a TASK_LOST event, we will only get the update during
+  // a reconciliation. In that case, the specific reason will be shadowed by REASON_RECONCILIATION.
+  // Since we don't know the original reason, we need to assume that the task might come back.
+  val MightComeBack: Set[mesos.Protos.TaskStatus.Reason] = Set(
+    mesos.Protos.TaskStatus.Reason.REASON_RECONCILIATION,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_DISCONNECTED,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_REMOVED
+  )
+
+  val WontComeBack: Set[mesos.Protos.TaskStatus.Reason] = {
+    mesos.Protos.TaskStatus.Reason.values().toSet.diff(MightComeBack)
+  }
+
+  trait Terminal
+
+  //scalastyle:off cyclomatic.complexity
+  def apply(taskStatus: mesos.Protos.TaskStatus): MarathonTaskStatus = {
```

Comment:
This one is also only needed for migration if I'm right – then this should be defined in that scope